### PR TITLE
[announce_routes] fix test_static_route_ecmp issue by announce_route optimization

### PR DESCRIPTION
**Pull request by StormLiangMS**
### Description of PR

Summary:
Fixes # (issue)
Fix the mistake that vlan ip has been announced to ToR DUT by exabgp, which could cause the route issue in DUT.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Fix the mistake that vlan ip has been announced to ToR DUT by exabgp, which could cause the route issue in DUT.

#### How did you do it?
Ignore the vlan ip when announcing routes.

#### How did you verify/test it?
Run test_static_route_ecmp

### CODE REVIEW

**Review by wangxin — APPROVED** [review]

**Review by Blueve — APPROVED** [review]

**Comment by Blueve** [comment]
@yaqiangz  for vis
@StormLiangMS Is this a day 0 bug in our annoucne_routes.py?

File: ansible/library/announce_routes.py [hunk 1]
```diff
@@ -206,8 +206,8 @@ def generate_routes(family, podset_number, tor_number, tor_subnet_number,
                     # Skip non podset 0 for T0
                     if podset != 0:
                         continue
-                    # Skip subnet 0 (vlan ip) for M0
-                    elif topo == "m0" and subnet == 0:
+                    # Skip subnet 0 (vlan ip)
+                    elif subnet == 0:
```
Comment (Blueve) [inline ansible/library/announce_routes.py:210]: I doubting this change will cause T1 topo miss first vlan route.

Comment (Blueve) [reply ansible/library/announce_routes.py:210]: Should we only add the condition for m0 and t0 topo? Only them have directly VLAN and require to skip DUT's vlan. For t1 topo, we shouldn't skip this subnet.

**Review by Blueve — CHANGES_REQUESTED** [review]
Let's hold this PR for a while. For t0 topo, we suppose announce route will not run branch.

**Comment by Blueve** [comment]
Since both t0 and lt0 have same vlan configuration, is possible the route were not learnt from exbgp?

I doubting that the duplicate route was learnt from this path:
```
lt0 ---BGP---> t1(ceos) --BGP--> t0
```

File: ansible/library/announce_routes.py (hunk 1 above)
Comment (yxieca) [reply ansible/library/announce_routes.py:210]: This is under route type == 'ToR"? (line 205), is it still a concern?

Comment (Blueve) [reply ansible/library/announce_routes.py:210]: The router type here is DUT's neighbors' type. For T1 DUT, it has tor neighbors and will go this branch.

**Pull request closed without merging**